packages/manager/apps/telecom/src/app/telecom/pack/move/offers/move-offers.controller.js

```diff
@@ -232,39 +233,79 @@ export default class PackMoveOffersCtrl {
   static updateSelectedGtrOption(offer, optionName) {
     let optionComfort = false;
     offer.options.forEach((option) => {
-      if (
-        option.name.startsWith('gtr_') &&
-        option.selected !== null &&
-        optionName.startsWith('gtr_')
-      ) {
-        if (option.name !== optionName) {
-          set(option, 'selected', false);
+      if (option.name.startsWith('gtr_') && option.selected !== null) {
+        if (optionName.startsWith('gtr_')) {
+          if (option.name !== optionName) {
+            set(option, 'selected', false);
+          } else {
+            set(option, 'selected', true);
+          }
+          if (optionName.match(/^gtr_\d{1,2}m_/) && option.selected === true) {
+            optionComfort = true;
+          }
+          set(offer, 'gtrComfortActivated', optionComfort);
         }
-        if (optionName.match(/^gtr_\d{1,2}m_/) && option.selected === true) {
-          optionComfort = true;
+        if (optionName === 'none') {
+          set(option, 'selected', false);
         }
-        set(offer, 'gtrComfortActivated', optionComfort);
       }
     });
   }
 
-  updateOfferPriceAndGtr(value, offer, optionName) {
-    this.constructor.updateSelectedGtrOption(offer, optionName);
-    this.constructor.updateOfferDisplayedPrice(value, offer, optionName);
+  updateOfferPriceAndGtr(offer) {
+    this.constructor.updateSelectedGtrOption(offer, offer.gtrSelected);
+    this.constructor.updateOfferDisplayedPrice(1, offer, offer.gtrSelected);
   }
 
   static isChosen(option) {
     return option.choosedValue > 0;
   }
 
-  static isGtrOption(optionName) {
-    return optionName.startsWith('gtr_');
-  }
-
   static isGtrOptionExist(offer) {
     return offer.options.some((option) => option.name.startsWith('gtr_'));
   }
 
+  initGtrOptions() {
+    this.offers = this.offers.map((offer) => {
+      // Initialize GTR options with a none value
+      let gtrOptions = [
+        {
+          label: this.$translate.instant('pack_move_no_gtr'),
+          name: 'none',
+        },
+      ];
+
+      const optionsMap = Object.values(offer.options)
+        .map((option) => {
+          if (option.name.startsWith('gtr_')) {
+            set(
+              option,
+              'label',
+              this.$translate.instant(`telecom_pack_migration_${option.name}`, {
+                price: option.optionalPrice.text,
+              }),
+            );
+            return option;
+          }
+          return undefined;
+        })
+        .filter((el) => el !== undefined);
```

Comment:
**nitpick:**  Since the need is to filter the options which starts with 'gtr' and then add a label. The combination of `filter + map` results in more readability than `map + filter`. Please try out this suggestion and it works.
```suggestion
        const gtrOptionsMap = Object.values(offer.options)
           .filter((el) => el.name.startsWith('gtr_'))
          .map((option) => ({
            ... option,
            label: this.$translate.instant(
                  `telecom_pack_migration_${option.name}`,
                  {
                    price: option.optionalPrice.text,
                  },
                )
          }));
```